unity/Assets/Scripts/CanOpen_Object.cs

```diff
@@ -97,529 +78,218 @@ void Start ()
-		{
-			//Interact();
-
-            //print("we are calling Reset() now");
-            //we are still open, trying to close, but hit something - reset to open
-			if(isOpen)
-			{
-				for (int i = 0; i < MovingParts.Length; i++)
-                {
-                    if (movementType == MovementType.Rotate)
-                    {
-						iTween.RotateTo(MovingParts[i], iTween.Hash(
-                        "rotation", openPositions[i] * currentOpenPercentage,
-                        "islocal", true,
-                        "time", animationTime,
-                        "easetype", "linear"));
-
-                    }
-
-
-                    else if (movementType == MovementType.Slide)
-                    {
-						iTween.MoveTo(MovingParts[i], iTween.Hash(
-                        "position", openPositions[i] * currentOpenPercentage,
-                        "islocal", true,
-                        "time", animationTime,
-                        "easetype", "linear"));
-
-                    }
-
-                    else if (movementType == MovementType.ScaleY)
-                    {
-                        iTween.ScaleTo(MovingParts[i], iTween.Hash(
-                        "scale", new Vector3(openPositions[i].x, closedPositions[i].y + (openPositions[i].y - closedPositions[i].y) * currentOpenPercentage, openPositions[i].z),
-                        "islocal", true,
-                        "time", animationTime,
-                        "easetype", "linear"));
-                    }
-
-                }
-			}
-
-			else
-			{
-				for (int i = 0; i < MovingParts.Length; i++)
-                {
-                    if (movementType == MovementType.Rotate)
-                    {
-						iTween.RotateTo(MovingParts[i], iTween.Hash(
-                        "rotation", closedPositions[i],
-                        "islocal", true,
-                        "time", animationTime,
-                        "easetype", "linear"));
-
-                    }
-
-
-                    else if (movementType == MovementType.Slide)
-                    {
-						iTween.MoveTo(MovingParts[i], iTween.Hash(
-                        "position", closedPositions[i],
-                        "islocal", true,
-                        "time", animationTime,
-                        "easetype", "linear"));
-                    }
-
-
-                    else if(movementType == MovementType.ScaleY)
-                    {
-                        iTween.ScaleTo(MovingParts[i], iTween.Hash(
-                        "scale", closedPositions[i],
-                        "islocal", true,
-                        "time", animationTime,
-                        "easetype", "linear"));
-                    }
-                }
-			}
-
+    // note: reset can interrupt the Interact() itween call because
+    // it will start a new set of tweens before onComplete is called from Interact()... it seems
+    public void Reset() {
+        if (!canReset) {
```

Comment:
I think this the correct reading. The animating object is reset to the `startOpenness` when it hits something. The `canReset` toggle itself is so that upon multiple instances of `OnTrigger` being called, it won't constantly try and reset back and forth forever in an endless loop of trying to resolve to the "last position."

So to that end, the object is only able to reset again once it has fully reset, hence `canReset`

Reply:
or maybe something like `hasReset` might be easier to understand?

Reply:
Or perhaps something like `isCurrentlyResetting`.

Reply:
used `isCurrentlyResetting` here